src/Passes/Source/RuleTransformationPass/Configuration.hpp

```diff
@@ -0,0 +1,109 @@
+#pragma once
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "Commandline/ConfigurationManager.hpp"
+
+namespace microsoft
+{
+namespace quantum
+{
+    /// Configuration class for the RuleTransformation pass.
+    class RuleTransformationPassConfiguration
+    {
+      public:
+        // Setup and construction
+        //
+
+        /// Setup function that attached the configuration to the ConfigurationManager.
+        void setup(ConfigurationManager& config);
+
+        /// Creates a configuration where all functionality is disabled.
+        static RuleTransformationPassConfiguration disable();
+
+        // Configuration classes
+        //
+
+        /// Testing whether this is the default configuration.
+        bool isDisabled() const;
+
+        /// Tests whether all functionality is disabled for this component.
+        bool isDefault() const;
+
+        // Properties
+        //
+
+        /// Whether or not the component should delete dead code.
+        bool deleteDeadCode() const;
```

Comment:
The fragment `deleteDeadCode` sounds like a command to change the state (imperative mood). But the comment and the implementation tell that this is a getter.
Consider renaming to a name more typical for getters, e.g. "**getD**eleteDeadCode()" or "deleteDeadCode**Flag**()".

Reply:
Fixed.